packages/cursorless-vscode/src/keyboard/grammar/getAcceptableTokenTypes.ts

```diff
@@ -43,39 +66,80 @@ function getMetadata<T extends keyof KeyboardCommandHandler>(
     .metadata;
 }
 
+// Prevent infinite recursion
```

Comment:
oof. why is 3 the right number?

(an alternative to the work queue and counts is also to use a fast/slow walk, like floyd warshall.)

Reply:
It's not, just picked something that felt likely to be an upper bound on what we might reasonably get in practice; updated comment